dashboard/app/controllers/api/v1/schools_controller.rb

```diff
@@ -7,4 +9,14 @@ def index
     end
     render json: serialized_schools
   end
+
+  # GET /dashboardapi/v1/schoolsearch/:criteria/:limit
+  def search
+    schools = School.where("name LIKE ?", "%#{params[:q].upcase}%").order(:name).limit(params[:limit])
```

Comment:
The `upcase` isn't necessary, our DB is case insensitive.
```
mysql> SHOW CREATE TABLE schools;
| Table   | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | schools | CREATE TABLE `schools` (
...
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci |
```
The `ci` in the collation denotes `case-insensitive`.